### PR TITLE
chore: codeSandbox error fixed in Select, SimpleSelect, Selectable

### PR DESCRIPTION
**Pull request by ToMESSKa**
Closes: INSTUI-4201

test plan:

- test if examples in Select, SimpleSelect, Selectable are opened without error in CodeSandbox

### CODE REVIEW

File: packages/__docs__/src/CodeSandboxButton/index.tsx [hunk 1]
```diff
@@ -73,14 +73,13 @@ import 'moment/min/locales'
...
-    const codeBlock = shouldRender
-      ? `render(${this.props.code})`
-      : this.props.code
+    const cleanedCode = this.props.code.replace(/\s+/g, '')
+    const renderEndMatcher = /render\(<.*>\)$/
+    const endsWithRenderStatement = renderEndMatcher.test(cleanedCode)
+    const codeBlock = endsWithRenderStatement
+      ? this.props.code
+      : `render(${this.props.code})`
+
```
Comment (ToMESSKa) [inline packages/__docs__/src/CodeSandboxButton/index.tsx:82]: The old code split the code into lines and checked if the last line included 'render(<'. However, sometimes the last line was just a ')' despite the code ending in a render function. This resulted in multiple render functions in CodeSandbox. Now, all newline characters and extra spaces are removed from the code, and a regex is used to check whether the code actually ends with 'render(<...>)', for example, 'render(<Example />)' or 'render(<div><Component /></div>)'.

File: packages/__docs__/src/CodeSandboxButton/index.tsx [hunk 2]
```diff
@@ -73,14 +73,15 @@ import 'moment/min/locales'
...
-      : this.props.code
+    const cleanedCode = this.props.code.replace(/\s+/g, '')
+    const lastRenderIndex = cleanedCode.lastIndexOf('render(<')
+    const endsWithRenderStatement =
+      lastRenderIndex !== -1 && cleanedCode.endsWith('>)')
+
+    const codeBlock = endsWithRenderStatement
+      ? this.props.code
+      : `render(${this.props.code})`
+
```
Comment (ToMESSKa) [inline packages/__docs__/src/CodeSandboxButton/index.tsx:84]: The old code split the code into lines and checked if the last line included 'render(<'. However, sometimes the last line was just a ')' despite the code ending in a render function. This resulted in multiple render functions in CodeSandbox. Now, all newline characters and extra spaces are removed from the code, and it is checked whether the code actually ends with render(<...>) like 'render(<Example />)' or 'render(<View><Component /></View>)

File: packages/__docs__/src/CodeSandboxButton/index.tsx [hunk 3]
```diff
@@ -73,14 +73,15 @@ import 'moment/min/locales'
...
 
-    const codeLines = this.props.code.split('\n')
-    const shouldRender = !codeLines[codeLines.length - 1]
-      .split(' ')
-      .join('')
-      .includes('render(<')
-    const codeBlock = shouldRender
-      ? `render(${this.props.code})`
-      : this.props.code
+    const cleanedCode = this.props.code.replace(/\s+/g, '')
```
Comment (matyasf) [inline packages/__docs__/src/CodeSandboxButton/index.tsx:76]: regexp are so hard to read, please add here a comment on what this one does

**Review by matyasf — CHANGES_REQUESTED** [review]
see my comment

File: packages/__docs__/src/CodeSandboxButton/index.tsx (hunk 3 above)
Comment (ToMESSKa) [reply packages/__docs__/src/CodeSandboxButton/index.tsx:77]: comment added

**Review by matyasf — APPROVED** [review]

**Review by balzss — APPROVED** [review]

**Pull request merged**